### ISSUE TITLE
Schema does not support operation type \"query\"

### ISSUE THREAD

**Issue by Southclaws** [99designs/gqlgen#2365]
### What happened?

Trying to use the embedded playground and I'm just getting 422 unprocessable entity:

```json
{"errors":[{"message":"Schema does not support operation type \"query\"","locations":[{"line":2,"column":5}],"extensions":{"code":"GRAPHQL_VALIDATION_FAILED"}}],"data":null}
```

### What did you expect?

Operation `query` to work.

### Minimal graphql.schema and models to reproduce

https://github.com/Southclaws/storyden

Fire this up to reproduce, visit http://localhost:8000/graphql/ui for the graphiql interface that's failing.

The setup code for the graphql server is in [app/transports/graphql/graphql.go](https://github.com/Southclaws/storyden/blob/main/app/transports/graphql/graphql.go)

### versions
 - `github.com/99designs/gqlgen v0.17.16`
 - 1.19 latest

**Comment by chaitanyapantheor** [comment]
@Southclaws i guess its not able to resolve on path `/query`

do something like this in your server file
`mux.Handle("/query", s)` [here](https://github.com/Southclaws/storyden/blob/main/app/transports/graphql/graphql.go#L42)

**Comment by rootkea** [comment]
No, this is due to `Query` being not there in schema. Try adding dummy `type Query` in https://github.com/Southclaws/storyden/blob/main/api/thread.graphqls and you'll see the error getting vanished.

This seems to be a bug in schema validator. FWIW, this error message originates from `gqlparser` package.

**Comment by Southclaws** [comment]
Ah this makes sense, thanks!

**Comment by rootkea** [comment]
Please let's not close this before this get fixed. Why should there be a compulsion to have `Query` in the schema?

**Comment by kantum** [comment]
I'm making an atomic API with only some mutation, I would like to be able to have no query at all.

**Comment by jakequade-hsm** [comment]
Late to the party - but this can also be due to your `glqgen.yml` file. I had to change/add these two parts:

```yml

federation:
  filename: ./internal/gql/generated/federation.go
  package: generated

model:
  filename: ./internal/gql/gqlmodel/models_gen.go
  package: gqlmodel
```
